File: CS/CustomWizardExample/Wizard/Presenters/ChooseDataSourceTypePageEx.cs
using System;
using CustomWizardExample.Wizard.Model;
using DevExpress.DataAccess.Wizard;
using DevExpress.DataAccess.Wizard.Presenters;
using DevExpress.DataAccess.Wizard.Views;

namespace CustomWizardExample.Wizard.Presenters {
    public class ChooseDataSourceTypePageEx : ChooseDataSourceTypePage<Model.CustomReportModel, IDataSourceWizardClient> {
        public ChooseDataSourceTypePageEx(IChooseDataSourceTypePageView view, IDataSourceWizardClient client) : base(view, client) { }

        public override Type GetNextPageType() {
            if(View.DataSourceType == CustomDataSourceType.Persons)
                return typeof(ChoosePersonsDataSourceTypePage);
            return base.GetNextPageType();
        }
    }
}

File: VB/CustomWizardExample/Wizard/Presenters/ChooseDataSourceTypePageEx.vb
Imports System
Imports CustomWizardExample.Wizard.Model
Imports DevExpress.DataAccess.Wizard
Imports DevExpress.DataAccess.Wizard.Presenters
Imports DevExpress.DataAccess.Wizard.Views

Namespace CustomWizardExample.Wizard.Presenters
	Public Class ChooseDataSourceTypePageEx
		Inherits ChooseDataSourceTypePage(Of Model.CustomReportModel, IDataSourceWizardClient)

		Public Sub New(ByVal view As IChooseDataSourceTypePageView, ByVal client As IDataSourceWizardClient)
			MyBase.New(view, client)
		End Sub

		Public Overrides Function GetNextPageType() As Type
			If View.DataSourceType Is CustomDataSourceType.Persons Then
				Return GetType(ChoosePersonsDataSourceTypePage)
			End If
			Return MyBase.GetNextPageType()
		End Function
	End Class
End Namespace
